### ISSUE TITLE
[META] ToDo (Recommendations)

### ISSUE THREAD

**Issue by mjmadsen** [PokemonGoF/PokemonGo-Bot#4004]
### Documentation
- [ ] Updated documentation. #4283
- [ ] Create updated FAQ for frequent issues. 
- [ ] Update wiki. 

### Anti-detection
- [x] Detect softban and stop. #3902
- [ ] GPS altitude to match longitude and latitude. #2762, #3621, #3808, #4114
- [x] Use random altitudes in the mean time, until above is completed.
- [ ] Device information variation. #3977 
- [ ] Variable sensor data. #3621
- [x] Configurable limitation on Pokemon caught per day. #3922
- [ ] Configurable limitation on stops spun per day. #3922
- [ ] Configurable limitation on distance traveled per day. #3922
- [ ] Configurable limitation on xp gained per day. #3922 #4261
- [ ] Configurable limitation on pokemon transfered per day. #3922 #4285
- [ ] Configurable limitation on pokemon evolved per day. #3922 #4285
- [x] Add a chance of missing in Pokemon catch. #3808 PR #4107
- [ ] Randomizing task order. 
- [ ] Walk between last recorded coord and new coord (when changed).  #3808 

### Features 
- [ ] Limit travel range to a particular area.  #3808

### Efficiency
- [ ] Traveling salesman AI applied to path-finding. 
- [ ] Reduce unneeded parameters for variables in NameSpace of self/bot. 

### Config efficiency
- [ ] Reduce config examples and merge project lines.  #4281
- [ ] Segregate Config to module based #3506 
- [ ] Add multiple accounts per config. Switch when above-mentioned limitations are met. 
- [ ] Detect config changes on the fly #3588
- [ ] Disabling/enabling other tasks at runtime #3858 
- [ ] "Class" of Pokemon for config.json #3980

### API (not this project directly)
- [ ] Complete "non-essential" API request/response analysis (any remaining unknowns)
- [ ] Further analysis to match human packets vs our generated packets.

**Comment by GitPupper** [comment]
Fixed your grammar. ;P

**Comment by mjmadsen** [comment]
My BAC might have been above legal limits during the initial post. Thanks @GitPupper

**Comment by k4n30** [comment]
I was just about to create such a post to link all the issues and feature requests. Are you able to link the relevant feature requests to your checklist for the devs to come to a single source please.  All of the individual requests and issues need to stay in their own threads and shouldn't be discussed here though, and items on the checklist should be marked as done, as they are completed.

Are you going to maintain this list?

**Comment by mjmadsen** [comment]
I can do that, but I might need assistance linking to issues and PRs.

**Comment by k4n30** [comment]
@mjmadsen I'm happy to keep rolling with this for now (I have edit rights on your post anyway). If we need to move it to a central location or a new position I'll close this issue then.  I'll help where I can but the find feature is your friend :p

@jrjhealey - another candidate for the wiki if we can use it

**Comment by Nik0lashka** [comment]
"salesman AI" yay!

**Comment by HecticSeptic** [comment]
Though not very explicit in the API section of OP: #3621 (sensor data) and #3977 (device info) are quite important anti-ban measures which should probably be considered as much more important than the other items, in addition to the already added exp/stop/catch limits (#3922 ). 

Especially since it's possible that Niantic would use these for upcoming ban waves (even on accounts which only botted briefly, or long ago).

**Comment by mjmadsen** [comment]
@HecticSeptic Added those to the list. Thank you!

**Comment by danielsdian** [comment]
I think the documentation of the config file is pretty bad regarding commands, options and etc. Im working on improving the Windows user experience.

**Comment by danielsdian** [comment]
We made the News in Brazil btw. Some tech specialized website talked about how people got lvl 40 using bot while must players arent over lvl 30. http://www.tecmundo.com.br/pokemon-go/107799-malandragem-jogador-usa-bot-alcancar-nivel-maximo-pokemon-go.htm

**Comment by HecticSeptic** [comment]
Some of those anti-detection checkboxes need checking now, right?

**Comment by mjmadsen** [comment]
@HecticSeptic Updated what I can see as done. Some were started and reverted and we're just waiting on a new PR. Some also were partially met and have been broke into different tasks. Please let me know if I have missed anything.

**Comment by MZorzy** [comment]
+ pokemongo-map deprecated as his readme.
think to update to other poke scanner/radar integration? or move here a fork

**Comment by Nik0lashka** [comment]
@MZorzy this is working fork https://github.com/PokemonGoMap/PokemonGo-Map

**Comment by mjmadsen** [comment]
I haven't updated this in a hell of a long time. @k4n30 I'm closing for now.
